lib/streamlit/navigation/page.py

```diff
@@ -300,7 +302,11 @@ def run(self) -> None:
                 module = types.ModuleType("__page__")
                 # We want __file__ to be the path to the script
                 module.__dict__["__file__"] = self._page
-                exec(code, module.__dict__)
+                if code.co_flags & CO_COROUTINE:
+                    # The source code includes top-level awaits, so the compiled code object is a coroutine.
+                    return eval(code, module.__dict__)
```

Comment:
Returns an awaitable only when the page contains top-level awaits so that the API keeps compatible with the original Streamlit in the case of not awaited pages even if the runtime AST patching doesn't work well.